src/components/Carousel.tsx

```diff
@@ -1,26 +1,223 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import './Carousel.scss';
 
-const Carousel: React.FC = () => (
-  <div className="Carousel">
-    <ul className="Carousel__list">
-      <li>
-        <img src="./img/1.png" alt="1" />
-      </li>
-      <li>
-        <img src="./img/1.png" alt="2" />
-      </li>
-      <li>
-        <img src="./img/1.png" alt="3" />
-      </li>
-      <li>
-        <img src="./img/1.png" alt="4" />
-      </li>
-    </ul>
-
-    <button type="button">Prev</button>
-    <button type="button">Next</button>
-  </div>
-);
+interface ImageCollection {
+  images: string[];
+  step: number;
+  frameSize: number;
+  itemWidth: number;
+  animationDuration: number;
+  infinite: boolean;
+}
+
+const Carousel: React.FC<ImageCollection> = ({
+  images,
+  step,
+  frameSize,
+  itemWidth,
+  animationDuration,
+  infinite,
+}) => {
+  const [previousWidth, setPreviousWidth] = useState<number>(itemWidth);
+
+  const [styleList, setStyleList] = useState({
+    transform: 'translateX(0)',
+    transition: `transform ${animationDuration / 1000}s ease-in-out`,
```

Comment:
you can just save the numbers here as values of the keys
```suggestion
    transform: 0,
    transition: animationDuration / 1000,
```